### PR TITLE
Add a simple VLAN plugin

### PR DESCRIPTION
**Pull request by dcbw**

### CODE REVIEW

File: plugins/vlan/vlan.go [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+
+func loadConf(bytes []byte) (*NetConf, string, error) {
+	n := &NetConf{}
+	if err := json.Unmarshal(bytes, n); err != nil {
+		return nil, "", fmt.Errorf("failed to load netconf: %v", err)
+	}
+	if n.Master == "" {
+		return nil, "", fmt.Errorf(`"master" field is required. It specifies the host interface name to create the VLAN for.`)
+	}
+	if n.VlanId < 0 || n.VlanId > 4095 {
```
Comment (squeed) [inline plugins/vlan/vlan.go]: IIRC, the highest allowed is 4094 / 0xFFE - 4095 is reserved.

File: plugins/vlan/vlan.go [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+	if err != nil {
+		return err
+	}
+
+	result.DNS = n.DNS
+
+	return types.PrintResult(result, cniVersion)
+}
+
+func cmdDel(args *skel.CmdArgs) error {
```
Comment (squeed) [inline plugins/vlan/vlan.go]: It looks like this delete isn't idempotent.

**Comment by dcbw** [comment]
@squeed PTAL, thanks!

File: plugins/vlan/vlan.go (hunk 2 above)
Comment (dcbw) [inline plugins/vlan/vlan.go]: fixed.

File: plugins/vlan/vlan.go (hunk 1 above)
Comment (dcbw) [inline plugins/vlan/vlan.go]: Fixed.

**Comment by squeed** [comment]
Looks good to me. Thanks for filling out more of the scaffolding.

**Comment by dcbw** [comment]
Thanks, just gonna merge it :)

**Pull request merged**